src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -470,25 +475,36 @@ private static boolean isInitializationSequence(
                 case TokenTypes.EXPR:
                     final DetailAST methodCallAst = currentSiblingAst.getFirstChild();
 
-                    if (methodCallAst.getType() == TokenTypes.METHOD_CALL) {
-                        final String instanceName =
-                            getInstanceName(methodCallAst);
-                        // method is called without instance
-                        if (instanceName.isEmpty()) {
+                    final FullIdent methodCallFullIdent =
+                        FullIdent.createFullIdentBelow(methodCallAst);
+                    final DetailAST methodCallIdent = methodCallFullIdent.getDetailAst();
+                    final boolean isMethodCalledFromNewObject =
+                        methodCallIdent.getParent().getType() == TokenTypes.LITERAL_NEW;
+
+                    if (methodCallAst.getType() == TokenTypes.METHOD_CALL
+                        && !isMethodCalledFromNewObject) {
+                        final String methodOrInstanceName =
+                            getMethodOrInstanceName(methodCallFullIdent);
+                        final boolean isMethodCalledWithInstance =
+                            isMethodCalledWithInstance(methodCallIdent);
```

Comment:
```suggestion
                        final boolean isMethodCalledWithInstance =
                                methodCallIdent.getParent().getType() == TokenTypes.DOT;
```

Reply:
That won't be logically correct, I looped my way up top as we can have `TYPECAST` as the parent before we can get to `DOT`, multiple typecasts are also possible, here is a simple example-
```java
((subClass)obj).setVal1(1);
```
Related AST-
```
 |--EXPR -> EXPR
 |   `--METHOD_CALL -> (
 |       |--DOT -> .
 |       |   |--LPAREN -> (
 |       |   |--TYPECAST -> (
 |       |   |   |--TYPE -> TYPE
 |       |   |   |   `--IDENT -> subClass
 |       |   |   |--RPAREN -> )
 |       |   |   `--IDENT -> obj -------> This has parent as typecast
 |       |   |--RPAREN -> )
 |       |   `--IDENT -> setVal1
 |       |--ELIST -> ELIST
 |       |   `--EXPR -> EXPR
 |       |       `--NUM_INT -> 1
 |       `--RPAREN -> )
 |--SEMI -> ;
```

> Additionally, it might help to refactor this method a bit, as a separate commit before adding new code.

That is a good point, there is some logical simplification that is possible here, I will send a minor PR for that.

Reply:
> That won't be logically correct, I looped my way up top…

Do we have a test case like this?

Reply:
@nick-mancuso Thinking on your suggestion, all method calls without instance, their `IDENT` always has `METHOD_CALL` as its parent, so the reverse condition worked.
```java
final boolean isMethodCalledWithInstance =
                            methodCallIdent.getParent().getType() != TokenTypes.METHOD_CALL;
```
But the problem is still the same, if I negate `!=` to `==` then also all the test cases will pass, branches are unique in that case too :(. Interesting mutation, any ideas?